etc/eslint/rules/typescript.js

```diff
@@ -0,0 +1,2426 @@
+*
+* @example
+* // Bad...
+* const self = this;
+* const that = this;
+*
+* // Good...
+* function example() {
+*   console.log(this);
+* }
+*/
+rules[ '@typescript-eslint/no-this-alias' ] = 'error';
+
+/**
+* Disables the rule against unnecessary boolean literal comparisons.
+*
+* @name no-unnecessary-boolean-literal-compare
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [no-unnecessary-boolean-literal-compare]{@link https://typescript-eslint.io/rules/no-unnecessary-boolean-literal-compare}
+*
+* @example
+* // Okay...
+* const x = true;
+* if ( x === true ) {
+*     // ...
+* }
+*/
+rules[ '@typescript-eslint/no-unnecessary-boolean-literal-compare' ] = 'off';
+
+/**
+* Disallows unnecessary qualifiers (like namespaces or enum names) when the referenced item is accessible without them.
+*
+* @name no-unnecessary-qualifier
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unnecessary-qualifier]{@link https://typescript-eslint.io/rules/no-unnecessary-qualifier}
+*
+* @example
+* // Bad...
+* namespace A {
+*     export const B = 1;
+* }
+* const x = A.B;
+*
+* // Good...
+* const B = 1;
+* const x = B;
+*/
+rules[ '@typescript-eslint/no-unnecessary-qualifier' ] = 'error';
+
+/**
+* Disallows unnecessary type arguments in generics where the argument is the default.
+*
+* @name no-unnecessary-type-arguments
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unnecessary-type-arguments]{@link https://typescript-eslint.io/rules/no-unnecessary-type-arguments}
+*
+* @example
+* // Bad...
+* const x: Array<number> = [];
+*
+* // Good...
+* const x: number[] = [];
+*/
+rules[ '@typescript-eslint/no-unnecessary-type-arguments' ] = 'error';
+
+/**
+* Disallows unnecessary type assertions that do not change the type of an expression.
+*
+* @name no-unnecessary-type-assertion
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unnecessary-type-assertion]{@link https://typescript-eslint.io/rules/no-unnecessary-type-assertion}
+*
+* @example
+* // Bad...
+* const x: number = 10;
+* const y = x as number;
+*
+* // Good...
+* const x: number = 10;
+* const y = x;
+*/
+rules[ '@typescript-eslint/no-unnecessary-type-assertion' ] = 'error';
+
+/**
+* Disallows unused expressions which have no effect on the state of the program.
+*
+* @name no-unused-expressions
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unused-expressions]{@link https://typescript-eslint.io/rules/no-unused-expressions}
+*
+* @example
+* // Bad...
+* 0;
+* if ( false ) {}
+*
+* // Good...
+* (() => {
+*     const a = 0;
+* })();
+*/
+rules[ '@typescript-eslint/no-unused-expressions' ] = 'error';
+
+/**
+* Disables the rule against unused variables.
+*
+* @name no-unused-vars
+* @memberof rules
+* @type {string}
+* @default [ 'error', { 'args': 'after-used' } ]
+* @see [no-unused-vars]{@link https://typescript-eslint.io/rules/no-unused-vars}
+*
+* @example
+* // Okay...
```

Comment:
@Planeshifter I believe this example needs updating, correct? `after-used` applies to function arguments. You could use something like

```
// Bad...
function foo( x, y, z ) {
    return x + y;
}

// Okay...
function bar( x, y, z ) {
    return y + z;
}
```

Reply:
Yes. Fixed.